svix-cli/src/relay/mod.rs

```diff
@@ -0,0 +1,467 @@
+            Err(_timeout_hit) => {
+                // Generous. 1.5x the ping frequency. If we go that long without
+                // seeing anything from the server, force a reconnect.
+                if last_msg.elapsed() > SERVER_PING_PERIOD + (SERVER_PING_PERIOD / 2) {
+                    anyhow::bail!(REMOTE_SERVER_CLOSED);
+                }
+            }
+            // Stream empty/closed
+            Ok(None) => break,
+            Ok(Some(msg)) => {
+                last_msg = Instant::now();
+
+                let data = match msg? {
+                    // Control messages.
+                    Message::Close(_) => anyhow::bail!(REMOTE_SERVER_CLOSED),
+                    Message::Ping(_) | Message::Pong(_) | Message::Frame(_) => continue,
+
+                    // Messages that carry data we care to process.
+                    Message::Text(s) => s.into(),
+                    Message::Binary(bytes) => bytes,
+                };
+
+                handle_incoming_message(client.clone(), data, &local_url, tx.clone()).await;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn send_to_ws_loop(
+    mut rx: UnboundedReceiver<MessageOut>,
+    mut tx: SplitSink<S, Message>,
+) -> Result<()> {
+    while let Some(msg) = rx.recv().await {
+        tokio::time::timeout(
+            WRITE_WAIT,
+            tx.send(Message::Binary(
+                serde_json::to_vec(&msg)
+                    .expect("trivial serialization")
+                    .into(),
+            )),
+        )
+        .await?
+        .context("Websocket write timeout")?;
+    }
+
+    Ok(())
+}
+
+async fn make_local_request(
+    client: HttpClient,
+    url: &url::Url,
+    data: MessageInEvent,
+) -> Result<LocalServerResponse> {
+    let method = data.method.parse()?;
+    // FIXME: deprecation warning
+    #[allow(deprecated)]
+    let body = base64::decode(&data.body)?;
+    let mut headers = HeaderMap::with_capacity(data.headers.len());
+    for (k, v) in &data.headers {
+        // FIXME: there's a remark about the Go client freaking out if there's more than one host header set.
+        //   Do we care now that we're not using Go? TBD.
+        headers.insert(
+            HeaderName::try_from(k.as_str())?,
+            HeaderValue::try_from(v.as_str())?,
+        );
+    }
+    Ok(client
+        .request(method, url.clone())
+        .timeout(DEFAULT_TIMEOUT)
+        .body(body)
+        .headers(headers)
+        .send()
+        .await?)
+}
+
+fn format_resp_headers(headers: &HeaderMap) -> Result<HashMap<String, String>> {
+    let mut out = HashMap::new();
+    const SKIP_USER_AGENT: HeaderValue = HeaderValue::from_static("Go-http-client/1.1");
+    for (k, v) in headers {
+        // FIXME: the Go cli used to unset the useragent if it matched the Go http client. Do we need something similar here? idk why this is needed.
+        if k == http::header::USER_AGENT && v == SKIP_USER_AGENT {
```

Comment:
Haha, I hope we don't actually need this?

Reply:
It's not clear to me what problem we were aiming to solve by removing this particular value in the first place. I'll remove it (since it clearly doesn't make sense today). If/when whatever problem shows up again, we can re-solve it 🤣